src/basic/nulstr-util.c

```diff
@@ -105,9 +105,11 @@ int strv_make_nulstr(char * const *l, char **ret, size_t *ret_size) {
                         return -ENOMEM;
 
                 n = 0;
-        } else
+        } else {
                 /* Make sure there is a second extra NUL at the end of resulting nulstr (not counted in return size) */
                 m[n] = '\0';
+                m[n+1] = '\0';
```

Comment:
This is not necessary, as in the above, string with trailing NUL is already copied. So, now we have three NUL at the end...